tests/core/pyspec/eth2spec/test/electra/block_processing/test_process_consolidation_request.py

```diff
@@ -562,6 +562,34 @@ def test_switch_to_compounding_with_pending_consolidations_at_limit(spec, state)
 # Tests that should fail
 
 
+@with_electra_and_later
+@with_presets([MINIMAL], "need sufficient consolidation churn limit")
+@with_custom_state(
+    balances_fn=scaled_churn_balances_exceed_activation_exit_churn_limit,
+    threshold_fn=default_activation_threshold,
+)
+@spec_test
+@single_phase
+def test_incorrect_same_source_target(spec, state):
+    # move state forward SHARD_COMMITTEE_PERIOD epochs to allow for consolidation
+    state.slot += spec.config.SHARD_COMMITTEE_PERIOD * spec.SLOTS_PER_EPOCH
+
+    # Set up an otherwise correct consolidation
+    current_epoch = spec.get_current_epoch(state)
+    source_index = spec.get_active_validator_indices(state, current_epoch)[0]
+    # Set source and target to be the same
+    target_index = source_index
+    source_address = b"\x22" * 20
+    set_compounding_withdrawal_credential_with_balance(spec, state, target_index, address=source_address)
```

Comment:
Can we add a comment here stating the importance of this line? Eg without it this would be treated as a switch to compounding request.

Reply:
done!